core/yaml.adoc

```diff
@@ -0,0 +1,125 @@
+= Yaml support for WildFly
+:author:            Emmanuel Hugonnet
+:email:             ehugonne@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+The aim of this proposal is to be able to inject external configuration in a YAML format to WildFly in order to customize an existing configuration.
+It is out of the scope of this proposal to provide support to define a complete server configuration in YAML: as such you won't be able to add a new extension using a YAML file.
+We also want this to be *idempotent*: that means that starting and restarting the server with the same command line should produce the same result and shouldn't fail.
+This feature is expected to be used as base for Bootable JAR implementation, it is not expected to be called directly by user (TODO get confirmation on this from Jeff and Brian and on how to 'hide' the feature).
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE[WFCORE-XXXX]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY[WFLY-XXXX]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+* mailto:jdenise@redhat.com[Jean-François Denise]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+WildFly will be able to take a list of YAML files from the command line to redefine its exisiting configuration.
+The operations will be applied in the order the files are defined and after the initial operations defined by the XML configuration.
+The YAML configuration should be *idempotent* so that we can start and restart with the same command line without failure nor resulting with a different final configuration.
```

Comment:
I'm not sure to understand what that means. If my YAML is adding a resource to a list, is it correct?

Does the persisted management model includes the resources from the YAML? If that's not the case, every time we start the server, it will start from a clean state and we can add the resource to the list?

Reply:
No I was thinking of having several YAML files as arguments.
The persisted management model can be the one updated after applying the YAML, resetting an attribute value to the same value would don nothing, since the resources would already be created the add wouldn't occur, same for the  delete.